### PR TITLE
[IMPROVE] Improve error listener, print the source line

### PR DESCRIPTION
**Pull request by xbsd-nikolay**

### CODE REVIEW

**Comment by xbsd-nikolay** [comment]
Nice stderr formatting

```
line 1:2 token recognition error at: '9'
{{9hello name="Dan" color = 'blue'}}
  ^

line 2:2 token recognition error at: '9'
{{9121hello name="Dan" color = 'blue'}}
  ^

line 2:3 token recognition error at: '1'
{{9121hello name="Dan" color = 'blue'}}
   ^

line 2:4 token recognition error at: '2'
{{9121hello name="Dan" color = 'blue'}}
    ^

line 2:5 token recognition error at: '1'
{{9121hello name="Dan" color = 'blue'}}
     ^
```

File: src/test/scala/de/zalando/beard/parser/BeardTemplateParserSpec.scala [hunk 1]
```diff
@@ -165,12 +165,30 @@ class BeardTemplateParserSpec extends FunSpec with Matchers {
...
+            Text("<hello>"),
+            NewLine(1),
+            AttrInterpolation(
+              Identifier("hello"),
+              Seq(AttributeWithValue("name", "Dan"), AttributeWithValue("color", "blue"))),
+            NewLine(1),
+            Text("</hello>"))))
+      }
+
       //        it("should not allow special chars inside of attribute identifiers") {
```
Comment (danpersa) [inline src/test/scala/de/zalando/beard/parser/BeardTemplateParserSpec.scala]: looking at the commented test... maybe you should uncomment it, and test that it gives a nice error also ;)

Comment (xbsd-nikolay) [inline src/test/scala/de/zalando/beard/parser/BeardTemplateParserSpec.scala]: It's a little bit tricky :) The Lexer show the error in the correct way, but we have stacktrace from the Parser. 

```
line 1:3 token recognition error at: '!'
{{h!el!l%o name="Dan" color = 'blue'}}
   ^

line 1:6 token recognition error at: '!'
{{h!el!l%o name="Dan" color = 'blue'}}
      ^

line 1:7 no viable alternative at input 'ell'

head of empty list
java.util.NoSuchElementException: head of empty list
    at scala.collection.immutable.Nil$.head(List.scala:420)
    at scala.collection.immutable.Nil$.head(List.scala:417)
```

Comment (xbsd-nikolay) [inline src/test/scala/de/zalando/beard/parser/BeardTemplateParserSpec.scala]: e.g. scala >

``` scala
scala> val h!el!l%o = "asdfa"
<console>:11: error: not found: value !
       val h!el!l%o = "asdfa"
               ^
<console>:11: error: not found: value !
       val h!el!l%o = "asdfa"
            ^
<console>:11: error: not found: value %
       val h!el!l%o = "asdfa"
```

Comment (danpersa) [inline src/test/scala/de/zalando/beard/parser/BeardTemplateParserSpec.scala]: if you're not able to test it, remove the test ;)

Comment (xbsd-nikolay) [inline src/test/scala/de/zalando/beard/parser/BeardTemplateParserSpec.scala]: I think it's better to keep the test, I've just changed it to except an exception :)

**Comment by danpersa** [comment]
:+1:

**Pull request merged**